packages/app-builder-lib/src/publish/KeygenPublisher.ts

```diff
@@ -113,4 +113,35 @@ export class KeygenPublisher extends HttpPublisher {
     const { account, product, platform } = this.info
     return `Keygen (account: ${account}, product: ${product}, platform: ${platform}, version: ${this.version})`
   }
+
+  // Get the filetype from a filename. Returns a string of one or more file extensions,
+  // e.g. `zip`, `dmg`, `tar.gz`, `tar.bz2`, `exe.blockmap`. We'd use `path.extname()`,
+  // but it doesn't support multiple extensions, e.g. `dmg.blockmap`.
+  private getFiletype(filename: string): string {
```

Comment:
What are your thoughts on extracting this to: https://github.com/electron-userland/electron-builder/blob/master/packages/app-builder-lib/src/util/sanitizeFileName.ts
Renaming that file to just be `util/filename.ts`?

And then we need some unit tests on this. Particularly because I'm horrible at working with regexes, but also I foresee some bug reports due to custom naming/macro expansions that might occur with artifact names.

Reply:
Ah, I didn't know custom naming was possible. Maybe this isn't the best solution then, to avoid any future issues. Do you think it'd better to use a whitelist then?

```ts
function getFiletype(filename: string): string {
  let extname = path.extname(filename)

  // Append leading extension for known compressed extensions
  switch (extname) {
    case '.blockmap':
    case '.bz2':
    case '.gz':
    case '.lz':
      extname = path.extname(filename.replace(extname, '')) + extname

      break
  }

  return extname
}
```

Probably much clearer, since it doesn't use regex. 🙂

Reply:
Moved to a util function and added unit tests, but still unsure of how to run them. I'm not super familiar with Jest but anytime I try to run tests, Jest doesn't run anything. Still trying to figure that one out. 😅